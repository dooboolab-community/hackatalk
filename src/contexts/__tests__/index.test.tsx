import * as React from 'react';

import { Text, View } from 'react-native';

import { StateProvider } from '..';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

let props = {
  navigation: {
    navigate: jest.fn(),
  },
};

describe('[AppProvider] rendering test', () => {
  let json: renderer.ReactTestRendererJSON;
  const component = (
    <StateProvider>
      <View>
        <Text>Placeholding Children</Text>
      </View>
    </StateProvider>
  );

  it('component and snapshot matches', () => {
    json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('[StateProvider] interactions', () => {
  let rendered: renderer.ReactTestRenderer;
  let root: renderer.ReactTestInstance;
  const component = (
    <StateProvider>
      <View>
        <Text>Placeholding Children</Text>
      </View>
    </StateProvider>
  );

  const user = {
    displayName: 'dooboolab',
    age: 30,
    job: '',
  };

  beforeEach(() => {
    props = {
      navigation: {
        navigate: jest.fn(),
      },
    };
    rendered = renderer.create(component);
    root = rendered.root;
  });

  // it('should trigger [resetUser] action', () => {
  //   let instance = root.instance;
  //   instance.actions.resetUser();
  // });

  // it('should check trigger actions when method called', () => {
  //   let instance = root.instance;
  //   instance.actions.setUser({
  //     displayName: '',
  //     age: 0,
  //     job: '',
  //   });
  // });
});
