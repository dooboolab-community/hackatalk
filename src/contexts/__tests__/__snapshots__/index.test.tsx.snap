// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AppProvider] rendering test component and snapshot matches 1`] = `
<View>
  <Text>
    Placeholding Children
  </Text>
</View>
`;
